instant-xml/src/lib.rs

```diff
@@ -53,7 +69,181 @@ to_xml_for_type!(i32);
 to_xml_for_type!(String);
 
 pub trait FromXml<'xml>: Sized {
-    fn from_xml(input: &str) -> Result<Self, Error>;
+    fn from_xml(_input: &str) -> Result<Self> {
+        unimplemented!();
+    }
+
+    fn deserialize<D>(deserializer: &mut D) -> Result<Self>
+    where
+        D: DeserializeXml<'xml>;
+
+    fn deserialize_attr<D>(_deserializer: &mut D, _value: &str) -> Result<Self>
+    where
+        D: DeserializeXml<'xml>,
+    {
+        unimplemented!();
+    }
+}
+
+pub trait DeserializeXml<'xml>: Sized {
+    fn deserialize_bool<V>(&mut self, _visitor: V) -> Result<V::Value>
+    where
+        V: Visitor<'xml>,
+    {
+        unimplemented!();
+    }
+
+    fn deserialize_struct<V>(
+        &mut self,
+        _visitor: V,
+        _name: &str,
+        _prefixes: &HashMap<&'xml str, &'xml str>,
+    ) -> Result<V::Value>
+    where
+        V: Visitor<'xml>,
+    {
+        unimplemented!();
+    }
+
+    fn deserialize_attribute<V>(&mut self, _visitor: V, _value: &str) -> Result<V::Value>
+    where
+        V: Visitor<'xml>,
+    {
+        unimplemented!();
+    }
+}
+
+pub trait Visitor<'xml>: Sized {
+    type Value;
+
+    fn visit_str(self, _value: &str) -> Result<Self::Value> {
+        unimplemented!();
+    }
+
+    fn visit_struct<'a, D>(
+        &self,
+        _deserializer: &mut D,
+        _attributes: Option<&HashMap<String, String>>,
+    ) -> Result<Self::Value>
+    where
+        D: DeserializeXml<'xml> + AccessorXml<'xml>,
+    {
+        unimplemented!();
+    }
+}
+
+pub trait AccessorXml<'xml> {
+    fn peek_next_tag(&mut self) -> Result<Option<XmlRecord>>;
+    fn verify_prefix(&self, prefix_to_verify: &str) -> bool;
+}
+
+pub struct Deserializer<'xml> {
+    pub iter: &'xml mut XmlParser<'xml>,
```

Comment:
This should probably own both `iter` (which should probably be called `parser`) and `prefixes`.

Reply:
I changed name to parser. Those `prefixes` (now `namespaces`) are from struct definition, not the part of the parser. The `tag_attributes` could probably be moved to parser, I will take a fresh look at it on Monday.